python/cudf/cudf/core/column/decimal.py

```diff
@@ -29,6 +29,11 @@ class DecimalColumn(ColumnBase):
     def __truediv__(self, other):
         return self.binary_operator("div", other)
 
+    def __setitem__(self, key, value):
+        if isinstance(value, np.integer):
```

Comment:
Do we want to use `cudf.utils.dtypes.is_integer` here instead?

Reply:
Is this special casing handling numpy scalars?

Reply:
yeah, this is just for numpy integers. Pyarrow doesn't like them for some reason.

```
>>> pa.scalar(1, type=pa.decimal128(1,0))
<pyarrow.Decimal128Scalar: Decimal('1')>
>>> pa.scalar(np.int64(1), type=pa.decimal128(1,0))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pyarrow/scalar.pxi", line 851, in pyarrow.lib.scalar
  File "pyarrow/error.pxi", line 107, in pyarrow.lib.check_status
pyarrow.lib.ArrowTypeError: int or Decimal object expected, got numpy.int64
```